src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_FindReferences_Legacy.cs

```diff
@@ -20,14 +20,22 @@ public static partial class SymbolFinder
         /// <param name="symbol">The symbol to find references to.</param>
         /// <param name="solution">The solution to find references within.</param>
         /// <param name="cancellationToken">A cancellation token.</param>
-        public static async Task<IEnumerable<ReferencedSymbol>> FindReferencesAsync(
+        public static Task<IEnumerable<ReferencedSymbol>> FindReferencesAsync(
             ISymbol symbol,
             Solution solution,
             CancellationToken cancellationToken = default)
+        {
+            return FindReferencesAsync(new SymbolAndProjectId(symbol, projectId: null), solution, cancellationToken);
+        }
+
+        internal static async Task<IEnumerable<ReferencedSymbol>> FindReferencesAsync(
+            SymbolAndProjectId symbolAndProjectId,
```

Comment:
> More of a curiosity thing, but is this essentially a deprecation of the other API?

Essentially yes.  But the other API is public.  So what can we do?

Note: this isn't about metadata/source per-se.  This is about OOP.  For OOP to work, it needs to serialize symbols to/from OOP/VS.  And to do that, it needs a project-context to be able to resolve the symbol back with an appropriate compilation.

If we don't pass the project-id around, we'll end up doing FAR in proc instead of in teh nice OOP process.

Reply:
Alright it was the OOP case context that I was missing because at least in-proc it didn't seem necessary.

Reply:
yup.!